tests/geometry_utils_tests/segments_tests.py

```diff
@@ -0,0 +1,52 @@
+from operator import itemgetter
+
+import pytest
+from hypothesis import (given,
+                        note)
+from lz.functional import compose
+from lz.iterating import capacity
+from shapely.geometry import (LinearRing,
+                              LineString)
+
+from pode.geometry_utils import segments
+from tests.strategies import linear_rings
+
+
+@given(linear_rings)
+def test_numbers_of_points_in_each_segment(ring: LinearRing) -> None:
+    note(f"Ring: {ring.wkt}")
+    assert all(len(line.coords) == 2 for line in segments(ring))
+
+
+@given(linear_rings)
+def test_number_of_segments(ring: LinearRing) -> None:
+    note(f"Ring: {ring.wkt}")
+    assert capacity(segments(ring)) == len(ring.coords) - 1
+
+
+@given(linear_rings)
+def test_first_and_last_points_equality(ring: LinearRing) -> None:
```

Comment:
I think it will be easier to not add `_equality` or any suffixes like this to test names, it makes test names too long and verbose, in case of assertions change it will be easy to forget to update test name as well, `test_end_points` should be enough IMO